includes/core/classes/class-setup.php

```diff
@@ -111,6 +113,37 @@ protected function setup_hooks(): void {
 		);
 	}
 
+	/**
+	 * Load plugin textdomain.
+	 *
+	 * Make sure to load all translation fileseven for users
+	 * who chosed another than the sites default language.
+	 *
+	 * @see https://developer.wordpress.org/reference/functions/load_plugin_textdomain
+	 *
+	 * Calling load_plugin_textdomain() should be delayed until init action.
+	 * @see https://developer.wordpress.org/reference/functions/load_plugin_textdomain/#comment-1568
+	 *
+	 * load_plugin_textdomain() will try to load the mo file firstly from:
+	 * WP_LANG_DIR . ‘/plugins/’ . $mofile
+	 * Only if it is not able to it will load it from GATHERPRESS_DIR_NAME folder
+	 * @see https://developer.wordpress.org/reference/functions/load_plugin_textdomain/#comment-3521
+	 *
+	 * Since WordPress 4.6 translations now take translate.wordpress.org as priority and so
+	 * plugins that are translated via translate.wordpress.org DO NOT necessary require load_plugin_textdomain() anymore.
+	 * If you don’t want to add a load_plugin_textdomain() call to your plugin
+	 * you have to set the Requires at least: field in your readme.txt to 4.6 or more.
+	 * @see https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#plugins-on-wordpress-org
+	 *
+	 * BUT, ...!!!
+	 *
+	 * A call to load_plugin_textdomain() is ABSOLUTELY NEEDED to properly load all translation files
+	 * for users who chosed another than the sites default language in their /profile.php.
+	 */
+	public function load_gatherpress_textdomain() {
```

Comment:
You really don't need this. It's pointless and only has downsides. And saying "A call to load_plugin_textdomain() is ABSOLUTELY NEEDED" is purely **WRONG**.

The just-in-time translation loading in WordPress handles everything for you and only loads the translations when needed.

Reply:
Hmm... a little bit strange.

I absolutely trust you in translation stuff, because I know you have been advocating and coding a lot for that topic.

From my long doc-comment you can imagine that I read & tried a lot to come up with my solution and the UPPERCASE comment. But unfortunately I can not make this work, without that line of code. Or maybe I have the wrong expectations.

## How to test

0. Comment out `load_plugin_textdomain( 'gatherpress', false, false )`
1. Set site language to English in `wp-admin/options-general.php`
2. Set user prefered language to German in `wp-admin/profile.php`
3. Check that the Admin-UI is German now, except everything GatherPress 
4. Re-Enable `load_plugin_textdomain( 'gatherpress', false, false )`
5. Check that the Admin-UI is German now, including everything GatherPress

Reply:
The issue is that you are triggering just-in-time translation loading **too early**.

WordPress tries to load the translations automatically as soon as you call `__()`. In your case, this happens somewhere before the current user is loaded, thus the admin won't be in German.

With this manual `load_plugin_textdomain()` call you're just papering over this issue, causing WP to try to load the translations twice.

The proper fix is to initialize your plugin later. This in the main plugin file works:

```
add_action(
	'plugins_loaded',
	static function () {
		// Initialize setups.
		GatherPress\Core\Setup::get_instance();
	}
);

// Or simply:

add_action(
	'plugins_loaded',
	array( '\GatherPress\Core\Setup', 'get_instance' )
);
```

Reply:
Ok, thank you very much Pascal, once again your explanation is better than any docs (I know) about that.

I followed your advise:

1. Set `add_action( 'plugins_loaded', array( '\GatherPress\Core\Setup', 'get_instance' ));`
2. Disabled my attempt `// add_action( 'init', array( $this, 'load_gatherpress_textdomain' ), 0 ); `
3. And started looking for the anoying bug again
4. and was able to track it down until `Assets->localize()`, which calls
   `'timezoneChoices'  => Utility::timezone_choices(),` and seems to be the culprit. Disabling this line brings all translations back into shape; having this enabled messes things up.
5. Going on ...

Reply:
Hello @mauteri ,

do we really need all possible timezones 

1. in the frontend markup?
2. in the backend markup?
3. on every page load?

https://github.com/carstingaxion/gatherpress/blob/02358ab60318a7464e31865ffea0f3bb8c961809/includes/core/classes/class-assets.php#L306

Reply:
Isn't `Assets::localize()` only called in `Assets::add_global_object()` which is hooked to `wp_head`? So that should be late enough.

----

Aside: just adding a script tag like this is usually not a good idea. You should at least use `wp_print_inline_script_tag`. Even better, attach it to a specific script with `wp_add_inline_script`.

Reply:
https://github.com/user-attachments/assets/dd4685bc-dc0e-4cd2-911e-2d954a456e5e

Reply:
Found it: :tada: 

```php
$timezones_raw   = explode( PHP_EOL, wp_timezone_choice( 'UTC', get_user_locale() ) );  // i18n FIX!
```

Reply:
As expected, a change like to the load order will break things. At least the unit tests.

Reply:
> Isn't `Assets::localize()` only called in `Assets::add_global_object()` which is hooked to `wp_head`? So that should be late enough.

And on `admin_print_scripts`, which is the problematic trace.

> Aside: just adding a script tag like this is usually not a good idea. You should at least use `wp_print_inline_script_tag`. Even better, attach it to a specific script with `wp_add_inline_script`.

:+1:  I wanted to suggest the same, even that the relevant data - the timezones - are only used for the Event Date Block (#684)